Source/CarthageKit/Xcode.swift

```diff
@@ -189,10 +189,24 @@ extension BuildArguments: CustomStringConvertible {
 public func locateProjectsInDirectory(directoryURL: NSURL) -> SignalProducer<ProjectLocator, CarthageError> {
 	let enumerationOptions: NSDirectoryEnumerationOptions = [ .SkipsHiddenFiles, .SkipsPackageDescendants ]
 
+	let submodules = submodulesInRepository(directoryURL)
+		.map { return $0.path }
+		.collect()
+		.first()?.value ?? []
+
 	return NSFileManager.defaultManager().carthage_enumeratorAtURL(directoryURL.URLByResolvingSymlinksInPath!, includingPropertiesForKeys: [ NSURLTypeIdentifierKey ], options: enumerationOptions, catchErrors: true)
+		.filter { _, URL in
+			let pathRelativeToDirectoryURL = URL.absoluteString.substringFromIndex(directoryURL.absoluteString.endIndex)
+			for submodulePath in submodules {
+				if pathRelativeToDirectoryURL.hasPrefix(submodulePath) {
```

Comment:
`hasPrefix` really isn't a safe way to do this. (Consider if `pathRelativeToDirectoryURL` is `foo-bar` and `submodulePath` is `foo`.) You'd need to use `NSURL.pathComponents` or `NSString.pathComponents`.

It'd be nice to add an extension to `NSURL` to test if one URL is a descendent of another URL.